snapshot/v3_snapshot.go

```diff
@@ -57,6 +57,7 @@ type Manager interface {
 	Status(dbPath string) (Status, error)
 
 	// Restore restores a new etcd data directory from given snapshot file.
+	// It returns an error if specified data directory already exists.
```

Comment:
@gyuho nit but I guess it will be helpful to add the reason of error as you have mentioned in the issue i.e.  "It returns an error if specified data directory already exists to prevent overwriting existing WAL files, which is unsafe for most operations". Thanks!

Reply:
Other than that it looks great to clarify this with doc as your PR does.